### ISSUE TITLE
Formatter duplicates comments in closure

### ISSUE THREAD

**Issue by cician** [grain-lang/grain#1089]
I tried formatting the JSON library with latest git main and stumbled on a problem. Comments get duplicated in a specific case.

I've narrowed it down to this:

```grain
import String from "string"

let makePrettyPrintJSONWriter = String.forEachCodePoint(c => {
  // a
  void
}, "")
```
which formats to this:

```grain
import String from "string"

let makePrettyPrintJSONWriter = String.forEachCodePoint(c => {
// a
  // a
  void
}, "")
```

Here's the piece of the original code, which may add some context;
```grain
      String.forEachCodePoint(
        codePoint => {
          if (codePoint == 47) {
            if (format.escapeHTMLUnsafeSequences && prevCodePoint == 60) {
              //printer.write("\\/")
              addCharFromCodePoint(92, buffer)
              addCharFromCodePoint(47, buffer)
            } else {
              // otherwise just emit the slash as-is
              //printer.writeChar('/')
              addCharFromCodePoint(codePoint, buffer)
            }
          } else {
            emitCodePoint(codePoint)
          }

          prevCodePoint = codePoint
        },
        s,
      )
```
which formats to this:
```grain
      String.forEachCodePoint(codePoint => {

//printer.write("\\/")

// otherwise just emit the slash as-is
//printer.writeChar('/')
        if (codePoint == 47) {
          if (format.escapeHTMLUnsafeSequences && prevCodePoint == 60) {
            //printer.write("\\/")
            addCharFromCodePoint(92, buffer)
            addCharFromCodePoint(47, buffer)
          } else {
            // otherwise just emit the slash as-is
            //printer.writeChar('/')
            addCharFromCodePoint(codePoint, buffer)
          }
        } else {
          emitCodePoint(codePoint)
        }

        prevCodePoint = codePoint
      }, s)
```

Other than this I'm pleased with latest improvements to the formatter.

**Comment by ng-marcus** [comment]
Thanks for reporting, I'll take a look!

**Comment by marcusroberts** [comment]
Thanks spotted the problem and fixed it in #1090

**Comment by cician** [comment]
I can confirm PR #1090 fixes it in the original case as well. Thank you.
